Incorrect typings for polygon's client JS file.

When compiling a node typescript project using Polygon, I get the following error:

```
node_modules/@polygon.io/client-js/dist/main.d.ts:9:5 - error TS2411: Property ''X-Polygon-Edge-User-Agent'' of type 'string | undefined' is not assignable to 'string' index type 'string'.

9     'X-Polygon-Edge-User-Agent'?: string;
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/@polygon.io/client-js/dist/main.d.ts:9
```

The relevant part in `main.d.ts`:
```typescript
import * as websocket from 'websocket';

interface IPolygonQuery {
    [key: string]: string | number | boolean | undefined;
}
interface IPolygonEdgeHeaders extends Record<string, string> {
    'X-Polygon-Edge-ID': string;
    'X-Polygon-Edge-IP-Address': string;
    'X-Polygon-Edge-User-Agent'?: string;
}
declare type IHeaders = IPolygonEdgeHeaders | Record<string, string>;
interface IRequestInit extends RequestInit {
    headers?: IHeaders;
}
declare type IRequestOptions = IRequestInit;
declare type IGet = (path: string, query: IPolygonQuery, options: IRequestOptions) => Promise<any>;
```

It seems to be complaining that a `Record` type cannot have an optional key.

### package.json
```
    "@polygon.io/client-js": "^7.2.2",
    "typescript": "~4.9.3",
```

--- a/package.json
+++ b/package.json
@@ -6,6 +6,7 @@
   "types": "dist/main.d.ts",
   "type": "module",
   "exports": {
+    "types": "./dist/main.d.ts",
     "import": "./dist/main.js",
     "require": "./dist/main.cjs"
   },

--- a/src/rest/index.ts
+++ b/src/rest/index.ts
@@ -11,6 +11,7 @@ export * from "./reference/index.js";
 export * from "./options/index.js";
 export * from "./stocks/index.js";
 export * from "./indices/index.js";
+export * from "./transport/request.js"; // ensure types are exported
 
 export interface IRestClient {
   crypto: ICryptoClient;

--- a/src/rest/transport/request.ts
+++ b/src/rest/transport/request.ts
@@ -5,15 +5,15 @@ export interface IPolygonQuery {
   [key: string]: string | number | boolean | undefined;
 }
 
-export interface IPolygonEdgeHeaders extends Record<string, string> {
-   'X-Polygon-Edge-ID': string;
-   'X-Polygon-Edge-IP-Address': string;
+export interface IPolygonEdgeHeaders extends Partial<Record<string, string>> {
+   'X-Polygon-Edge-ID'?: string;
+   'X-Polygon-Edge-IP-Address'?: string;
    'X-Polygon-Edge-User-Agent'?: string;
 }
 
-export type IHeaders = IPolygonEdgeHeaders | Record<string, string>
+export type IHeaders = IPolygonEdgeHeaders | HeadersInit
 
-export interface IRequestInit extends RequestInit {
+export interface IRequestInit extends Omit<RequestInit, 'headers'> {
   headers?: IHeaders
 }
 

